packages/web3-common/src/types.ts

```diff
@@ -0,0 +1,37 @@
+export type JSONRPCId = string | number | null;
+export type JSONRPCResult = string | number | boolean | Record<string, unknown>;
+export type JSONRPCIdentifier = '2.0' | '1.0';
+
+export interface JSONRPCError<T = JSONRPCResult> {
+	readonly code: number;
+	readonly message: string;
+	readonly data?: T;
+}
+
+export interface JSONRPCResponseWithError<T = JSONRPCResult> {
+	readonly id: JSONRPCId;
+	readonly jsonrpc: JSONRPCIdentifier;
+	readonly error: JSONRPCError<T>;
+	readonly result?: never;
+}
+
+export interface JSONRPCResponseWithResult<T = JSONRPCResult> {
+	readonly id: JSONRPCId;
+	readonly jsonrpc: JSONRPCIdentifier;
+	readonly error?: never;
+	readonly result: T;
+}
+
+export type JSONRPCResponse<T = JSONRPCResult> =
```

Comment:
it should `export type JSONRPCResponse<T = JSONRPCResult> =  JSONRPCResponseWithError<T> | JSONRPCResponseWithResult<T>;`

Reply:
Yes its exactly same but formatted over multi-line syntax.